src/library_wasi.js

```diff
@@ -270,6 +270,7 @@ var WasiLibrary = {
 #else
   fd_write__deps: ['$printChar'],
 #endif
+  fd_write__proxy: 'none', // Each Worker can do stdout/stderr printing on its own, without needing to proxy to the main thread.
```

Comment:
I'm pretty sure `doWritev` won't work if you call it on a thread.   And I think its not true for `printChar` either since you would end up with a different `printCharBuffers` on each thread.. which is not correct.

Reply:
> you would end up with a different `printCharBuffers` on each thread.. which is not correct.

I looked into this some decade ago, and from back then, I could not find defined semantics on how multiple threads writing on stdout/stderr should synchronize, when fflush is not called by the user. In most implementations, like ours, printing `\n` is an implicit fflush.

If multiple threads race to print to stdout/stderr, then it does seem to me it would be better semantics for each thread to construct full \n buffered lines that they would then race to print, rather than each thread mixing their individual character writes that could result in the data from multiple threads inter-mixing with each other?

Each Worker has a copy of MEMFS/library_tty.js in them, so they would each end up calling to default_tty_ops.put_char() and pushing out console.log() calls independently?

Having each thread synchronously block to perform stdout prints would seem like a rough performance overhead to have?

Reply:
But the function is not only used for stdout/stderr, its used for all file access. 

If we do make this change it would be very large one (basically making the filesystem multi-threaded) and I would at least expect a separate RP for it.

Reply:
Err, you're right of course. I somehow was interpreting this from the perspective that this function was already just for the stdout/stderr writing case. That would not work. I'll think of something here.